pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -305,65 +456,58 @@ pub mod pallet {
 				None => Err(Error::<T>::MessagePackingNotStarted.into()),
 			}
 		}
-	}
-
-	impl<T: Config> Pallet<T> {
-		/// Give the message to the `InboundMessageHandler` to be processed.
-		fn process_inbound_message(
-			domain_address: DomainAddress,
-			message: T::Message,
-		) -> (DispatchResult, Weight) {
-			let mut count = 0;
-
-			for submessage in message.submessages() {
-				count += 1;
-
-				if let Err(e) = T::InboundMessageHandler::handle(domain_address.clone(), submessage)
-				{
-					// We only consume the processed weight if error during the batch
-					return (Err(e), LP_DEFENSIVE_WEIGHT.saturating_mul(count));
-				}
-			}
-
-			(Ok(()), LP_DEFENSIVE_WEIGHT.saturating_mul(count))
-		}
 
-		/// Retrieves the router stored for the provided domain, sends the
-		/// message using the router, and calculates and returns the required
-		/// weight for these operations in the `DispatchResultWithPostInfo`.
-		fn process_outbound_message(
-			sender: DomainAddress,
+		/// Manually increase the proof count for a particular message and
+		/// executes it if the required count is reached.
+		///
+		/// Can only be called by `AdminOrigin`.
+		#[pallet::weight(T::WeightInfo::execute_message_recovery())]
+		#[pallet::call_index(11)]
+		pub fn execute_message_recovery(
```

Comment:
@hieronx Given that with the new LP logic, routers are now stored under domains, we need an extra arg here. Is this OK?

Reply:
> routers are now stored under domains

Maybe I could create some confusion with this before, but a router already has the domain information on it (it comes from our latest conversations). So you would not need to store a router under domains.

In the map: `message_hash -> router_id -> (message, votes, session)` the `router_id` already takes care of the domain

Reply:
Please check the [routers storage](https://github.com/centrifuge/centrifuge-chain/blob/5c8059756c60d0f1d04b8b4e0f9bd55971850359/pallets/liquidity-pools-gateway/src/lib.rs#L190). 

My understanding was that we can have multiple sets of routers, if so, it made sense to me that the `Domain` is the one that differentiates them.

Reply:
Also `message_hash -> router_id -> (message, votes, session)` is not the layout for inbound messages. I think it makes more sense to store these under the `session_id` key, but let's discuss that separately.

Reply:
> My understanding was that we can have multiple sets of routers, if so, it made sense to me that the Domain is the one that differentiates them.

Or is it expected that we always have one set of routers regardless of bridge setups etc?

Reply:
I think an additional extrinsic arg is not an issue at all. Just the messages need to match

Reply:
> My understanding was that we can have multiple sets of routers, if so, it made sense to me that the Domain is the one that differentiates them.

https://github.com/centrifuge/centrifuge-chain/pull/1961#discussion_r1715099023